apps/server/src/modules/files-storage/client/s3-client.adapter.ts

```diff
@@ -183,6 +186,50 @@ export class S3ClientAdapter implements IStorageClient {
 		return this.client.send(req);
 	}
 
+	public async list(prefix: string, maxKeys = Infinity) {
+		this.logger.log({ action: 'list', params: { prefix, bucket: this.config.bucket } });
+
+		try {
+			let files: string[] = [];
+			let ret: ListObjectsV2CommandOutput | undefined;
+
+			do {
+				const req = new ListObjectsV2Command({
+					Bucket: this.config.bucket,
+					Prefix: prefix,
+					ContinuationToken: ret?.NextContinuationToken,
+					MaxKeys: Math.min(maxKeys - files.length, 1000),
```

Comment:
As far as I understood the docs, 1000 is the programmatic maximum for MaxKeys. So I think you don't have to set that manually here and this should give the same result:
```suggestion
					MaxKeys: maxKeys - files.length,
```